### PR TITLE
Adds support for emojis and more IDNA URLs

### PR DESCRIPTION
**Pull request by automationator**
I found that the existing URL validator did not fully support certain IDNA URLs or emojis in URLs. This adds support for these while continuing to pass all existing tests.

### CODE REVIEW

**Comment by kvesteri** [comment]
Can you fix the flake8 errors?

**Comment by automationator** [comment]
They're fixed now, sorry about that.

File: setup.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def get_version():
     install_requires=install_requires,
     build_requires=install_requires,
     extras_require=extras_require,
+    setup_requires=['wheel'],
```
Comment (kvesteri) [inline setup.py:56]: What is the purpose of this line? Can you remove this since it is unrelated to the changes you've made?

Comment (automationator) [reply setup.py]: It's removed now.
